src/furo/__init__.py

```diff
@@ -104,13 +104,28 @@ def _compute_hide_toc(context):
     return has_exactly_one_list_item(context["toc"])
 
 
+@lru_cache(maxsize=None)
+def _furo_asset_hash(path):
+    _static = "_static/"
+    if _static not in path:
+        raise ValueError("furo_asset_hash expect a path with '_static' in it")
```

Comment:
I'd prefer to not actually hash the contents of the file, especially if we're baking assumptions about the configuration. I'd rather have a per-compile unique hash, to identify that the asset has changed.

The main purpose it serves is to avoid hitting the cache when the CSS is refreshed by live-reload, which even status quo does well. I'd prefer to set:

```py
_asset_hash = secrets.token_hex(12)

...

def _html_page_context(...):
    ...
    # set context asset_hash to the global defined above.

Reply:
> The main purpose it serves is to avoid hitting the cache when the CSS is refreshed by live-reload, which even status quo does well. I'd prefer to set:

Sorry, I believe I don't understand the reason for which you want this hash here.

If I put a hash on an url is because I want to cache that resource unless the client has the need of fetch it again. Live reload is hip but _in production_ I don't want all the clients to reload all the static resources just because there's a deployment that didn't touch them. I mean, [this is not good](https://github.com/psycopg/psycopg.github.io/commit/711ea94a8d5b0ed3efe09ae4431cd8f1b5e3f6a1).

If it has to be a per-build random, can it be disabled for real buils?

Reply:
Fair enough. Consider me convinced. :)